tools/src/main/scala/scala/scalanative/build/LLVM.scala

```diff
@@ -31,16 +32,34 @@ private[scalanative] object LLVM {
    *  @return
    *    The paths of the `.o` files.
    */
-  def compile(config: Config, paths: Seq[Path]): Seq[Path] = {
+  def compile(config: Config, paths: Seq[Path])(implicit
+      incCompilationContext: IncCompilationContext = new IncCompilationContext(
+        config.workdir
+      )
```

Comment:
There's a total of 5 instantiations of `IncCompilationContext` as the default argument. I don't think it should be needed. We probably only should have a single in the `Build::build` method. If for some reason implicit argument would not available we risk losing the state loaded previously.